lib/combineNodes.js

```diff
@@ -0,0 +1,310 @@
+                mergeChildWithParent(gltf, rootId, childId, childIndex, preserveCameras);
+                children.splice(childIndex, 1);
+            }
+            else {
+                childIndex++;
+            }
+        }
+        root.children = children;
+    }
+}
+
+//Returns true if the child has the necessary properties to be combined with its parent.
+function canCombine(parent, child, preserveCameras) {
+    if (!child.extras._pipeline.isMergeable) {
+        //If the child is animated, merging the parent node may lose transformation information for the child.
+        parent.extras._pipeline.isMergeable = false;
+        return false;
+    }
+
+    if (defined(parent.skeletons) !== defined(child.skeletons) || (defined(parent.skeletons) && (!arrayEqual(parent.skeletons.sort(), child.skeletons.sort()))) ||
+            propertyNotShared(parent, child, 'skin') ||
+            propertyNotShared(parent, child, 'jointName')) {
+        return false;
+    }
+    if (preserveCameras && (parent.camera !== child.camera)) {
+        return false;
+    }
+
+    return true;
+}
+
+//Merges the child into the parent node, transforming the mesh data if needed.
+function mergeChildWithParent(gltf, parentId, childId, childIndex, preserveCameras) {
+    var nodes = gltf.nodes;
+    var parent = nodes[parentId];
+    var child = nodes[childId];
+    checkCombineWarning(child, 'node', childId);
+
+    var meshes = gltf.meshes;
+    //Duplicate and transform the child's mesh data if the node has a transformation, and add the mesh to the parent.
+    if (defined(child.meshes) && child.meshes.length > 0) {
+        var meshId = child.meshes[0];
+        var newMeshId = meshId;
+
+        var transform = child.matrix;
+        if (defined(transform)) {
+            transform = Matrix4.fromColumnMajorArray(transform);
+            if (!Matrix4.equals(transform, Matrix4.IDENTITY)) {
+                newMeshId += '_0';
+                var idCount = 0;
+                var meshKeys = Object.keys(meshes);
+                while (meshKeys.indexOf(newMeshId) !== -1) {
+                    idCount++;
+                    newMeshId = newMeshId.slice(0, -1) + idCount;
+                }
+
+                var mesh = gltf.meshes[meshId];
+                var newMesh = clone(mesh, true);
+                var primitives = newMesh.primitives;
+                if (defined(primitives)) {
+                    var primitivesLength = primitives.length;
+                    for (var i = 0; i < primitivesLength; i++) {
+                        var attributes = primitives[i].attributes;
+                        if (defined(attributes)) {
+                            transformAttribute(gltf, meshId, transform, 'POSITION', attributes, gltf.accessors, gltf.buffers, gltf.bufferViews);
+                            transformAttribute(gltf, meshId, transform, 'NORMAL', attributes, gltf.accessors, gltf.buffers, gltf.bufferViews);

```

Comment:
No, I will fix that! In that case, I will give transformAttribute three cases - one for positions, normals, and vectors.